data-products/src/main/scala/org/sunbird/obsrv/dataproducts/job/MasterDataProcessorIndexer.scala

```diff
@@ -0,0 +1,156 @@
+        return
+      }
+      val datasource = datasources.get.head
+      val paths = getPaths(datasource)
+      val events_count = createDataFile(dataset, paths.outputFilePath)
+      val ingestionSpec = updateIngestionSpec(datasource, paths.datasourceRef, paths.ingestionPath)
+      submitIngestionTask(ingestionSpec)
+      DatasetRegistry.updateDatasourceRef(datasource, paths.datasourceRef)
+      if (!datasource.datasourceRef.equals(paths.datasourceRef)) {
+        deleteDataSource(datasource.datasourceRef)
+      }
+      val end_time = System.currentTimeMillis()
+      val success_time = end_time - time
+      metrics.generate(ets = new DateTime(DateTimeZone.UTC).getMillis, datasetId = dataset.id, edata = Edata(metric = Map(metrics.getMetricName("success_dataset_count") -> 1, metrics.getMetricName("total_time_taken") -> success_time, metrics.getMetricName("total_events_processed") -> events_count), labels = List(MetricLabel("job", "MasterDataIndexer"), MetricLabel("datasetId", dataset.id), MetricLabel("cloud", s"${config.getString("cloudStorage.provider")}"))))
+    } catch {
+      case e: Exception =>
+        metrics.generate(ets = new DateTime(DateTimeZone.UTC).getMillis, datasetId = dataset.id, edata = Edata(metric = Map(metrics.getMetricName("failure_dataset_count") -> 1), labels = List(MetricLabel("job", "MasterDataIndexer"), MetricLabel("datasetId", dataset.id), MetricLabel("cloud", s"${config.getString("cloudStorage.provider")}")), err = "Failed to index dataset.", errMsg = e.getMessage))
+        e.printStackTrace()
+    }
+  }
+
+  def getPaths(datasource: DataSource): Paths = {
+    val dt = new DateTime(DateTimeZone.UTC).withTimeAtStartOfDay()
+    val timestamp = dt.getMillis
+    val date = dayPeriodFormat.print(dt)
+    val provider = providerFormat(config.getString("cloudStorage.provider"))
+    val cloudPrefix = if (config.getString("cloudStorage.provider") == "azure") {
+      provider.sparkProviderURIFormat + s"""://${config.getString("cloudStorage.accountName")}.blob.core.windows.net/${config.getString("cloudStorage.container")}/"""
+    } else {
+      provider.sparkProviderURIFormat + s"""://${config.getString("cloudStorage.container")}/"""
+    }
+    val pathSuffix = s"""masterdata-indexer/${datasource.datasetId}/$date/"""
+    val ingestionPath = cloudPrefix.replace(provider.sparkProviderURIFormat, provider.druidProviderPrefix) + pathSuffix
+    val datasourceRef = datasource.datasource + '-' + date
+    val outputFilePath = cloudPrefix + pathSuffix
+    Paths(datasourceRef, ingestionPath, outputFilePath, timestamp)
+  }
+
+  private def updateIngestionSpec(datasource: DataSource, datasourceRef: String, filePath: String): String = {
+    val deltaIngestionSpec = s"""{"type":"index_parallel","spec":{"dataSchema":{"dataSource":"$datasourceRef"},"ioConfig":{"type":"index_parallel"},"tuningConfig":{"type":"index_parallel","maxRowsInMemory":500000,"forceExtendableShardSpecs":false,"logParseExceptions":true}}}"""
```

Comment:
You could do the same thing here as well.

Reply:
okay